lib/controllers/Authentication/auth_screen_controller.dart

```diff
@@ -9,6 +9,14 @@ import 'package:wave/utils/enums.dart';
 class AuthScreenController extends ChangeNotifier {
   LOGIN loginState = LOGIN.IDLE;
   REGISTER registerState = REGISTER.IDLE;
+  bool _obscuredText = false;
```

Comment:
The current code uses a single instance of the obscuredText variable, which is shared between both the login and sign-up screens. Here are the specific points of concern:

Shared State: Sharing the obscuredText variable between the login and sign-up screens can lead to unintended side effects. Toggling the password visibility on one screen will also affect the other screen, which is not the desired behavior.

Encapsulation: Each screen should have its own independent state for managing password visibility. This ensures that the behaviour of one screen does not impact the other.

**Suggested Solution**

To address these issues, we should encapsulate the state within each screen. Here's a proposed refactoring:

Separate State Variables: Define separate obscuredText variables for each screen.

Independent Methods: Implement the togglePasswordVisibility method within each screen's state management.